server/routerlicious/packages/lambdas-driver/src/kafka-service/runner.ts

```diff
@@ -139,15 +139,14 @@ export class KafkaRunner implements IRunner {
 			if (!this.runnerMetric.isCompleted()) {
 				this.runnerMetric.error("Kafka runner encountered an error during stop", error);
 			}
-			if (caller === "sigterm") {
-				this.deferred?.resolve();
-			} else {
-				// uncaughtException
+			if (caller === "uncaughtException") {
```

Comment:
is there a reason we are switching back to adding this if condition? I thought we only wanted to treat sigterm differently and rest can fall under common else block which would include uncaughtexception?

Reply:
I am making this explicit to make it clear that error.uncaughtException would exist if restart reason is "uncaughtException".

Reply:
is there a case where caller could be something other than sigterm or uncaughtexception?

Reply:
it could be empty in the case of caught exceptions: https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services-shared/src/runner.ts#L35. In that case, deferred is already rejected from start() method.

Reply:
@sharmakhushboo please wait as I am updating this because the tests are failing due to this change.
There is this test https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas-driver/src/test/kafka-service/runner.spec.ts#L42 which expects that the runner.stop should resolve the promise after processing the messages.

I am going to update the if-else as : `if "uncaughtException" then reject, else resolve`. Any other exception is expected to be resolved/rejected from start() method itself. Let me know if you have any concerns.